crates/biome_configuration/src/javascript/mod.rs

```diff
@@ -100,6 +117,101 @@ impl FromStr for JsxRuntime {
     }
 }
 
+fn deserialize_optional_jsx_factory_from_string<'de, D>(
+    deserializer: D,
+) -> Result<Option<JsxFactory>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    match parse_jsx_factory(&s) {
+        Some(factory) => Ok(Some(factory)),
+        None => Err(serde::de::Error::custom(format!(
+            "expected valid identifier or qualified name, but received {s}"
+        ))),
+    }
+}
+
+fn parse_jsx_factory(value: &str) -> Option<JsxFactory> {
+    use biome_js_syntax::*;
+    let syntax = parse_module(value, JsParserOptions::default());
+    let item = syntax.try_tree()?.items().into_iter().next()?;
+    match item {
+        AnyJsModuleItem::AnyJsStatement(stmt) => {
+            match JsExpressionStatement::cast_ref(stmt.syntax())?
+                .expression()
+                .ok()?
+            {
+                AnyJsExpression::JsStaticMemberExpression(member) => {
+                    let mut expr = member.object().ok();
+                    while let Some(e) = expr {
+                        let syntax = e.into_syntax();
+                        if let Some(ident) = JsIdentifierExpression::cast_ref(&syntax) {
+                            return Some(JsxFactory(ident.text().to_owned()));
+                        } else if let Some(member) = JsStaticMemberExpression::cast_ref(&syntax) {
+                            expr = member.object().ok();
+                        } else {
+                            break;
+                        }
+                    }
+                }
+                AnyJsExpression::JsIdentifierExpression(ident) => {
+                    return Some(JsxFactory(ident.text().to_owned()));
+                }
+                _ => (),
+            }
+        }
+        _ => (),
+    }
+
+    None
+}
+
+#[derive(Bpaf, Clone, Debug, Deserialize, Eq, Merge, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase")]
+pub struct JsxFactory(pub String);
+
+impl Default for JsxFactory {
+    fn default() -> Self {
+        Self("React".to_string())
```

Comment:
`JsxFactory` only stores its namespace. We eagerly parse the expression to support configuration validation. 
Using `React` instead of `React.createElement` or `React.Fragment` to make it more performant.